create_aio_app/template/hooks/pre_gen_project.py

```diff
@@ -0,0 +1,22 @@
+import argparse
+import sys
+
+import click
+
+from create_aio_app.utils.config import name_type
+
+
+def validate_project_name(project_name: str) -> None:
+    try:
+        name_type(project_name)
+    except argparse.ArgumentTypeError as exc:
+        click.echo(str(exc), err=True)
+        sys.exit(1)
```

Comment:
why do u use that ` sys.exit(1)`?

Reply:
It is preffered way to force cookiecutter stop generation. https://cookiecutter.readthedocs.io/en/latest/advanced/hooks.html#exit-with-an-appropriate-status